This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Sun, 9 Nov 2025 13:45:34 +0000
Subject: [PATCH 1/2] Fix compiler warning-as-error

error: narrowing conversion of '(DWORD)expectedReadSize' from 'DWORD' {aka 'long unsigned int'} to 'qsizetype' {aka 'int'} [-Wnarrowing]

diff --git a/src/base/utils/os.cpp b/src/base/utils/os.cpp
index 1111111..2222222 100644
--- a/src/base/utils/os.cpp
+++ b/src/base/utils/os.cpp
@@ -345,7 +345,7 @@ bool Utils::OS::applyMarkOfTheWeb(const Path &file, const QString &url)
         ::GetFileSizeEx(handle, &streamSize) && (streamSize.QuadPart > 0))
     {
         const DWORD expectedReadSize = std::min<LONGLONG>(streamSize.QuadPart, 1024);
-        QByteArray buf {expectedReadSize, '\0'};
+        QByteArray buf {static_cast<int>(expectedReadSize), '\0'};
 
         if (DWORD actualReadSize = 0;
             ::ReadFile(handle, buf.data(), expectedReadSize, &actualReadSize, nullptr) && (actualReadSize == expectedReadSize))

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Sun, 9 Nov 2025 14:29:31 +0000
Subject: [PATCH 2/2] Fix link on mingw cross-compilation


diff --git a/src/base/CMakeLists.txt b/src/base/CMakeLists.txt
index 1111111..2222222 100644
--- a/src/base/CMakeLists.txt
+++ b/src/base/CMakeLists.txt
@@ -243,7 +243,7 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
         ${IOKit_LIBRARY}
     )
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    target_link_libraries(qbt_base PRIVATE Iphlpapi PowrProf)
+    target_link_libraries(qbt_base PRIVATE iphlpapi powrprof)
 endif()
 
 if (NOT GUI)
